service/javadsl/kafka/server/src/main/scala/com/lightbend/lagom/internal/javadsl/broker/kafka/JavadslRegisterTopicProducers.scala

```diff
@@ -62,27 +64,30 @@ class JavadslRegisterTopicProducers @Inject() (
               case tagged: TaggedOffsetTopicProducer[AnyRef, _] =>
                 val tags = tagged.tags.asScala.toIndexedSeq
 
-                val eventStreamFactory: (String, AkkaOffset) => Source[(AnyRef, AkkaOffset), _] = { (tag, offset) =>
-                  tags.find(_.tag == tag) match {
-                    case Some(aggregateTag) =>
-                      tagged
-                        .readSideStream(
-                          aggregateTag,
-                          OffsetAdapter.offsetToDslOffset(offset)
-                        )
-                        .asScala
-                        .map { pair =>
-                          pair.first -> OffsetAdapter.dslOffsetToOffset(pair.second)
-                        }
-                    case None => throw new RuntimeException("Unknown tag: " + tag)
-                  }
+                val eventStreamFactory: (String, AkkaOffset) => Source[(Message[AnyRef], AkkaOffset), _] = {
+                  (tag, offset) =>
+                    tags.find(_.tag == tag) match {
+                      case Some(aggregateTag) =>
+                        tagged
+                          .readSideStream(
+                            aggregateTag,
+                            OffsetAdapter.offsetToDslOffset(offset)
+                          )
+                          .asScala
+                          .map { pair =>
+                            pair.first -> OffsetAdapter.dslOffsetToOffset(pair.second)
+                          }
+                      case None => throw new RuntimeException("Unknown tag: " + tag)
+                    }
                 }
 
-                val partitionKeyStrategy: Option[AnyRef => String] = {
-                  val javaPKS = topicCall.properties().getValueOf(KafkaProperties.partitionKeyStrategy())
-                  if (javaPKS != null) {
-                    Some(message => javaPKS.computePartitionKey(message))
-                  } else None
+                val transform: Message[AnyRef] => ProducerRecord[String, AnyRef] = message => {
+                  val key = Option(topicCall.properties.getValueOf(KafkaProperties.partitionKeyStrategy())) match {
+                    case Some(javaPKS) => javaPKS.computePartitionKey(message.getPayload)
+                    case None          => null
```

Comment:
I don't think it's worth wrapping it into an Option just of avoid a `null` check. 

In the end, we are wrapping and unwrapping (creating unnecessary allocation) just for doing exactly the same as `if(!null) ... else ...`.

If we were using the `Option` further on I would agree, but that's not the case here.

Reply:
adjusted without Option